sphinx4-core/src/main/java/edu/cmu/sphinx/api/FeatureExtractor.java

```diff
@@ -0,0 +1,146 @@
+package edu.cmu.sphinx.api;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+import edu.cmu.sphinx.frontend.Data;
+import edu.cmu.sphinx.frontend.DataEndSignal;
+import edu.cmu.sphinx.frontend.DataProcessingException;
+import edu.cmu.sphinx.frontend.DoubleData;
+import edu.cmu.sphinx.frontend.FloatData;
+import edu.cmu.sphinx.frontend.FrontEnd;
+import edu.cmu.sphinx.frontend.util.StreamDataSource;
+import edu.cmu.sphinx.util.props.ConfigurationManager;
+
+/**
+ * Extracts features from input stream
+ * 
+ * @author Vladisav Jelisavcic
+ * 
+ */
+class FeatureExtractor {
+    private FrontEnd frontEnd;
+    private StreamDataSource audioSource;
+    private int featureLength = -1;
+
+    /** The logger for this class */
+    private static final Logger logger = Logger
+            .getLogger("edu.cmu.sphinx.api.FeatureExtractor");
+
+    /**
+     * Constructs a FeatureExtractor. 
+     * 
+     * @param cm
+     *            the configuration manager
+     * @param frontEndName
+     *            the name for the frontEnd to be used
+     * @param inputStream
+     * 			  data stream 
+     *            
+     * @throws IOException if error occurred
+     */
+    public FeatureExtractor(ConfigurationManager cm, String frontEndName, InputStream inputStream)
```

Comment:
Ok, but if I understand correctly, then the user can't choose the FrontEnd
configuration,
and it should always use the default xml configuration (default.config.xml)
(or we should hardcode the FrontEnd like in e.g. BandDetector)?


Thanks,
VJ



On Thu, Sep 10, 2015 at 1:57 PM, Nickolay V. Shmyrev <
notifications@github.com> wrote:

> In sphinx4-core/src/main/java/edu/cmu/sphinx/api/FeatureExtractor.java
> <https://github.com/cmusphinx/sphinx4/pull/37#discussion_r39151311>:
>
> > +    private static final Logger logger = Logger
> > +            .getLogger("edu.cmu.sphinx.api.FeatureExtractor");
> > +
> > +    /**
> > +     * Constructs a FeatureExtractor.
> > +     *
> > +     * @param cm
> > +     *            the configuration manager
> > +     * @param frontEndName
> > +     *            the name for the frontEnd to be used
> > +     * @param inputStream
> > +     * 			  data stream
> > +     *
> > +     * @throws IOException if error occurred
> > +     */
> > +    public FeatureExtractor(ConfigurationManager cm, String frontEndName, InputStream inputStream)
>
> We do not expose configuration managers in high-level API, it is better to
> make all those details private.
>
> Like I wrote in the original prototype, it is better to support different
> sample rates.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cmusphinx/sphinx4/pull/37/files#r39151311>.
>